raw-spans-grouper/src/main/java/org/hypertrace/core/rawspansgrouper/TraceEmitPunctuator.java

```diff
@@ -0,0 +1,137 @@
+package org.hypertrace.core.rawspansgrouper;
+
+import io.micrometer.core.instrument.Timer;
+import java.nio.ByteBuffer;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.apache.avro.generic.IndexedRecord;
+import org.apache.kafka.streams.processor.Cancellable;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.processor.PunctuationType;
+import org.apache.kafka.streams.processor.Punctuator;
+import org.apache.kafka.streams.processor.To;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.state.ValueAndTimestamp;
+import org.hypertrace.core.datamodel.RawSpan;
+import org.hypertrace.core.datamodel.RawSpans;
+import org.hypertrace.core.datamodel.StructuredTrace;
+import org.hypertrace.core.datamodel.TimestampRecord;
+import org.hypertrace.core.datamodel.Timestamps;
+import org.hypertrace.core.datamodel.shared.DataflowMetricUtils;
+import org.hypertrace.core.datamodel.shared.trace.StructuredTraceBuilder;
+import org.hypertrace.core.serviceframework.metrics.PlatformMetricsRegistry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Checks if a trace can be finalized and emitted based on inactivity period of {@link
+ * RawSpansGroupingTransformer#groupingWindowTimeoutMs}
+ */
+public class TraceEmitPunctuator implements Punctuator {
+
+  private static final Logger logger = LoggerFactory.getLogger(TraceEmitPunctuator.class);
+
+  private double dataflowSamplingPercent = -1;
+  private final String DATAFLOW_SAMPLING_PERCENT = "dataflow.metriccollection.sampling.percent";
```

Comment:
Can you check, seems un-used - DATAFLOW_SAMPLING_PERCENT?

Reply:
fixed